Pyc code warnings

```
/home/akochkov/radare/radare2/libr/..//libr/bin/p/bin_pyc.c: In function ‘symbols’:
/home/akochkov/radare/radare2/libr/..//libr/bin/p/bin_pyc.c:90:31: warning: passing argument 1 of ‘r_list_newf’ from incompatible pointer type [-Wincompatible-pointer-types]
   90 |  RList *shared = r_list_newf (r_list_free);
      |                               ^~~~~~~~~~~
      |                               |
      |                               void (*)(RList *) {aka void (*)(struct r_list_t *)}
In file included from /home/akochkov/radare/radare2/libr/include/r_util.h:10,
                 from /home/akochkov/radare/radare2/libr/include/r_bin.h:4,
                 from /home/akochkov/radare/radare2/libr/..//libr/bin/p/bin_pyc.c:3:
/home/akochkov/radare/radare2/libr/include/r_list.h:77:36: note: expected ‘RListFree’ {aka ‘void (*)(void *)’} but argument is of type ‘void (*)(RList *)’ {aka ‘void (*)(struct r_list_t *)’}
   77 | R_API RList *r_list_newf(RListFree f);
      |                          ~~~~~~~~~~^
gcc -c  -MD   -fPIC -g -Wall -D__UNIX__=1 -DR2_PLUGIN_INCORE -Iformat -Imangling -I/home/akochkov/radare/radare2/libr/../shlr/zip/include -I../format/pyc/ -I/home/akochkov/radare/radare2/libr/../shlr/zip/include -I/home/akochkov/radare/radare2/libr/..//shlr/yxml/ -I/home/akochkov/radare/radare2/libr -I/home/akochkov/radare/radare2/libr/include -fvisibility=hidden -o /home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.o /home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c
/home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c: In function ‘get_array_object_generic’:
/home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c:495:27: warning: passing argument 1 of ‘r_list_newf’ from incompatible pointer type [-Wincompatible-pointer-types]
  495 |  ret->data = r_list_newf (free_object);
      |                           ^~~~~~~~~~~
      |                           |
      |                           void (*)(pyc_object *)
In file included from /home/akochkov/radare/radare2/libr/include/r_io.h:6,
                 from /home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c:3:
/home/akochkov/radare/radare2/libr/include/r_list.h:77:36: note: expected ‘RListFree’ {aka ‘void (*)(void *)’} but argument is of type ‘void (*)(pyc_object *)’
   77 | R_API RList *r_list_newf(RListFree f);
      |                          ~~~~~~~~~~^
/home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c: In function ‘get_dict_object’:
/home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c:587:27: warning: passing argument 1 of ‘r_list_newf’ from incompatible pointer type [-Wincompatible-pointer-types]
  587 |  ret->data = r_list_newf (free_object);
      |                           ^~~~~~~~~~~
      |                           |
      |                           void (*)(pyc_object *)
In file included from /home/akochkov/radare/radare2/libr/include/r_io.h:6,
                 from /home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c:3:
/home/akochkov/radare/radare2/libr/include/r_list.h:77:36: note: expected ‘RListFree’ {aka ‘void (*)(void *)’} but argument is of type ‘void (*)(pyc_object *)’
   77 | R_API RList *r_list_newf(RListFree f);
      |                          ~~~~~~~~~~^
/home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c: In function ‘get_sections_symbols_from_code_objects’:
/home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c:1206:22: warning: passing argument 1 of ‘r_list_newf’ from incompatible pointer type [-Wincompatible-pointer-types]
 1206 |  refs = r_list_newf (free_object);
      |                      ^~~~~~~~~~~
      |                      |
      |                      void (*)(pyc_object *)
In file included from /home/akochkov/radare/radare2/libr/include/r_io.h:6,
                 from /home/akochkov/radare/radare2/libr/..//libr/bin/p/../format/pyc/marshal.c:3:
/home/akochkov/radare/radare2/libr/include/r_list.h:77:36: note: expected ‘RListFree’ {aka ‘void (*)(void *)’} but argument is of type ‘void (*)(pyc_object *)’
   77 | R_API RList *r_list_newf(RListFree f);
      |                          ~~~~~~~~~~^
```

--- a/libr/asm/arch/pyc/opcode.c
+++ b/libr/asm/arch/pyc/opcode.c
@@ -179,7 +179,7 @@ pyc_opcodes *new_pyc_opcodes() {
 		ret->opcodes[i].op_pop = 0;
 	}
 
-	ret->opcode_arg_fmt = r_list_newf (free);
+	ret->opcode_arg_fmt = r_list_newf ((RListFree)free);
 	return ret;
 }
 

--- a/libr/bin/format/pyc/marshal.c
+++ b/libr/bin/format/pyc/marshal.c
@@ -492,7 +492,7 @@ static pyc_object *get_array_object_generic(RBuffer *buffer, ut32 size) {
 	if (!ret) {
 		return NULL;
 	}
-	ret->data = r_list_newf (free_object);
+	ret->data = r_list_newf ((RListFree)free_object);
 	if (!ret->data) {
 		free (ret);
 		return NULL;
@@ -584,7 +584,7 @@ static pyc_object *get_dict_object(RBuffer *buffer) {
 	if (!ret) {
 		return NULL;
 	}
-	ret->data = r_list_newf (free_object);
+	ret->data = r_list_newf ((RListFree)free_object);
 	if (!ret->data) {
 		R_FREE (ret);
 		return NULL;
@@ -1203,7 +1203,7 @@ fail:
 bool get_sections_symbols_from_code_objects(RBuffer *buffer, RList *sections, RList *symbols, RList *cobjs, ut32 magic) {
 	bool ret;
 	magic_int = magic;
-	refs = r_list_newf (free_object);
+	refs = r_list_newf ((RListFree)free_object);
 	if (!refs) {
 		return false;
 	}

--- a/libr/bin/p/bin_pyc.c
+++ b/libr/bin/p/bin_pyc.c
@@ -65,7 +65,7 @@ static RList *sections(RBinFile *arch) {
 }
 
 static RList *entries(RBinFile *arch) {
-	RList *entries = r_list_newf (free);
+	RList *entries = r_list_newf ((RListFree)free);
 	if (!entries) {
 		return NULL;
 	}
@@ -87,16 +87,16 @@ static ut64 baddr(RBinFile *bf) {
 }
 
 static RList *symbols(RBinFile *arch) {
-	RList *shared = r_list_newf (r_list_free);
+	RList *shared = r_list_newf ((RListFree)r_list_free);
 	if (!shared) {
 		return NULL;
 	}
-	RList *cobjs = r_list_newf (free);
+	RList *cobjs = r_list_newf ((RListFree)free);
 	if (!cobjs) {
 		r_list_free (shared);
 		return NULL;
 	}
-	interned_table = r_list_newf (free);
+	interned_table = r_list_newf ((RListFree)free);
 	if (!interned_table) {
 		r_list_free (shared);
 		r_list_free (cobjs);
@@ -105,13 +105,13 @@ static RList *symbols(RBinFile *arch) {
 	r_list_append (shared, cobjs);
 	r_list_append (shared, interned_table);
 	arch->o->bin_obj = shared;
-	RList *sections = r_list_newf (free);
+	RList *sections = r_list_newf ((RListFree)free);
 	if (!sections) {
 		r_list_free (shared);
 		arch->o->bin_obj = NULL;
 		return NULL;
 	}
-	RList *symbols = r_list_newf (free);
+	RList *symbols = r_list_newf ((RListFree)free);
 	if (!symbols) {
 		r_list_free (shared);
 		arch->o->bin_obj = NULL;

